Server/ReasnAPI/ReasnAPI/Services/TagService.cs

```diff
@@ -119,6 +74,24 @@ public void DeleteTag(int tagId)
         context.SaveChanges();
     }
 
+    public void ForceDeleteTag(int tagId)
+    {
+        var tag = context.Tags.FirstOrDefault(r => r.Id == tagId);
+        if (tag is null)
+        {
+            throw new NotFoundException("Tag not found");
+        }
+
+        var eventsWithTags = context.Events.Include(e => e.Tags).ToList();
+        foreach (var eventWithTags in eventsWithTags)
+        {
+            eventWithTags.Tags.Remove(tag);
+        }
```

Comment:
And same as before, we don't want to get ALL events just to remove one tag, as tags are unique, you can manage query which return events only having this tag

Reply:
changed